console_win.go

```diff
@@ -496,79 +513,95 @@ func mrec2btns(mbtns, flags uint32) ButtonMask {
 }
 
 func (s *cScreen) getConsoleInput() error {
-	rec := &inputRecord{}
-	var nrec int32
-	rv, _, er := procReadConsoleInput.Call(
-		uintptr(s.in),
-		uintptr(unsafe.Pointer(rec)),
-		uintptr(1),
-		uintptr(unsafe.Pointer(&nrec)))
-	if rv == 0 {
-		return er
-	}
-	if nrec != 1 {
-		return nil
-	}
-	switch rec.typ {
-	case keyEvent:
-		krec := &keyRecord{}
-		krec.isdown = geti32(rec.data[0:])
-		krec.repeat = getu16(rec.data[4:])
-		krec.kcode = getu16(rec.data[6:])
-		krec.scode = getu16(rec.data[8:])
-		krec.ch = getu16(rec.data[10:])
-		krec.mod = getu32(rec.data[12:])
-
-		if krec.isdown == 0 || krec.repeat < 1 {
-			// its a key release event, ignore it
+	waitObjects := []syscall.Handle{s.in, s.cancelflag}
+
+	rv, _, er := procWaitForMultipleObjects.Call(
+		uintptr(len(waitObjects)),
+		uintptr(unsafe.Pointer(&waitObjects[0])),
+		uintptr(0),
+		^uintptr(0))
```

Comment:
This is *really* strange.  I understand what you're doing, which is to try to get to INFINITE, which is -1.  Would it be clearer to set INFINITE to some variable - say uintptr(-1) ?

This stuff is kind of gnarly, but anything we can do to make it a little less dense may help future generations -- or just future me. :-)